plugins/modules/site_workflow_manager.py

```diff
@@ -929,42 +1660,117 @@ def get_diff_deleted(self, config):
+
+            if not site_response:
+                self.delete_single_site(site_id, site_name)
+                return self
+
+            sorted_site_resp = sorted(
+                site_response, key=lambda x: x.get("groupHierarchy"), reverse=True)
+
+            for item in sorted_site_resp:
+                self.delete_single_site(item['id'], item['name'])
 
-        if len(site_response) == 0:
             self.delete_single_site(site_id, site_name)
-            return self
+            self.log(
+                "The site '{0}' and its child sites have been deleted successfully".format(site_name), "INFO")
 
-        # Sorting the response in reverse order based on hierarchy levels
-        sorted_site_resp = sorted(site_response, key=lambda x: x.get("groupHierarchy"), reverse=True)
+        elif self.get_ccc_version_as_integer() >= self.version_2_3_7_6:
+            site_params = self.want.get("site_params")
+            site_params["site_id"] = self.have.get("site_id")
+            site_type = site_params.get("type")
+            site_id = self.have.get("site_id")
 
-        # Deleting each level in reverse order till topmost parent site
-        for item in sorted_site_resp:
-            self.delete_single_site(item['id'], item['name'])
+            site_assigned_network_devices_response = self.dnac._exec(
+                family="site_design",
+                function="get_site_assigned_network_devices",
+                op_modifies=True,
+                params={"site_id": site_id},
+            )
+            self.log(
+                "Received API response from 'get_site_assigned_network_devices': {0}".format(str(site_assigned_network_devices_response)), "DEBUG")
+            site_response = site_assigned_network_devices_response.get(
+                "response", [])
+
+            if site_response:
+                self.module.fail_json(
+                    msg=(
+                        "Site '{0}' cannot be deleted because it has assigned devices. "
+                        "Please delete devices first then delete the site.".format(site_name)
+                    )
+                )
 
-        # Delete the final parent site
-        self.delete_single_site(site_id, site_name)
-        self.log("The site '{0}' and its child sites have been deleted successfully".format(site_name), "INFO")
+                return self
+
+            response = None
+            if site_type == "floor":
+                response = self.delete_floor(site_name)
+            elif site_type == "area":
+                response = self.delete_area(site_name)
+                self.log(
+                    "Manual test response for deleting area: {0}".format(str(response)), "DEBUG")
+            elif site_type == "building":
+                response = self.delete_building(site_name)
+
+            self.log(
+                "Received API response from 'deleted': {0}".format(str(response)), "DEBUG")
```

Comment:
Add the site name as well while logging the response so that it will clear for which site we are logging the response then only debugging will be easier.
```
"Received API response from deleted {0}: {1}".format(site_type, str(response)), "DEBUG")

Reply:
Modified this log